tests/test_vmec.py

```diff
@@ -1166,66 +1176,64 @@ def test(
             rtol=rtol_vmec_desc_wout,
         )
 
-    # R & Z & lambda
+    # R & Z
     test("rmn", "R", use_nyq=False)
-    test("zmn", "Z", use_nyq=False, atol_vmec_desc_wout=4e-2)
-
+    test("zmn", "Z", use_nyq=False, atol_vmec_desc_wout=1e-2)
+    test(
+        "lmn",
+        "lambda",
+        use_nyq=False,
+        atol_vmec_desc_wout=1e-2,
+        negate_DESC_quant=True,
+        grid=vol_half_grid,
+        is_half_grid=True,
+    )
     # |B|
     test("bmn", "|B|", rtol_desc_desc_wout=7e-4)
 
     # B^zeta
-    test("bsupvmn", "B^zeta")  # ,rtol_desc_desc_wout=6e-5)
+    test("bsupvmn", "B^zeta")
 
     # B_zeta
-    test("bsubvmn", "B_zeta", rtol_desc_desc_wout=3e-4)
+    test("bsubvmn", "B_zeta", grid=vol_half_grid, is_half_grid=True)
 
     # hard to compare to VMEC for the currents, since
     # VMEC F error is worse and equilibria are not exactly similar
     # just compare back to DESC
     test("currumn", "J^theta", atol_vmec_desc_wout=1e4)
     test("currvmn", "J^zeta", negate_DESC_quant=True, atol_vmec_desc_wout=1e5)
 
-    # can only compare lambda, sqrt(g) B_psi B^theta and B_theta at bdry
```

Comment:
Isn't this still true? Even with the same gauge freedom there's still no guarantee they end up with the same poloidal angle

Reply:
Yes but the particular solution I have, they end up finding similar lambdas